### ISSUE TITLE
should we actively use ssh -T -e none ?

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#207]
ssh has an escape feature, usually triggered by entering `<ENTER> + "~" + ...`.

as we use ssh to pipe binary data, the data stream could contain the escape sequence.
but it seems this feature only gets activated if a terminal (pty) is connected to the ssh process (but we use pipes).

so, the question is how save are the defaults / the detection of pty - or whether we should use "ssh -T -e none" to actively switch off pty allocation and escape char usage.

if we would like to do that, the next question is whether -T -e none is supported on all platforms by all ssh implementations we use.

**Comment by edgewood** [comment]
The OpenSSH source uses isatty() to determine whether input is from a 
terminal or not 
(https://github.com/openssh/openssh-portable/blob/master/ssh.c line 
1105).  isatty() is a POSIX function.  In glibc 
(https://sourceware.org/git/?p=glibc.git;a=tree), it's a wrapper around 
tcgetattr, which in turn is a call into the kernel, which would expect 
know for certain the difference between a pipe file descriptor and a 
TTY.

In short, ssh isn't using heuristics to determine TTY-ness, but rather 
asking an authoritative source, so I don't think it's worth it.

-- 
Ed Blackman

**Comment by ThomasWaldmann** [comment]
@edgewood thanks for checking this, I agree.
